src/AutoUpdate/TranslationStatus.php

```diff
@@ -15,6 +15,8 @@ class TranslationStatus {
 	 * @return int|null
 	 */
 	public static function get( WPML_Post_Element $element ) {
+		wpml_tm_load_element_translations();
```

Comment:
@strategio I don't think this is the right place for this. Adding it here makes it very hard to understand the dependencies.

Basically `$wpml_tm_element_translations` is required in `WPML_TM_Translation_Status` so I would move the initialisation to there.

```
	public function filter_translation_status( $status, $trid, $target_lang_code ) {
		/** @var WPML_TM_Element_Translations $wpml_tm_element_translations */
		$wpml_tm_element_translations = wpml_tm_load_element_translations();
```

Reply:
You are completely right @brucepearson (added in https://git.onthegosystems.com/wpml/wpml-translation-management/-/merge_requests/1656/diffs?commit_id=b2851e1f7acb74554dd337febda0404b9e79242f).